src/t_stream.c

```diff
@@ -3119,20 +3166,38 @@ void xclaimCommand(client *c) {
     mstime_t deliverytime = -1;  /* -1 means IDLE/TIME options not given. */
     int force = 0;
     int justid = 0;
+    int mkgroup = 0;
 
-    if (o) {
-        if (checkType(c,o,OBJ_STREAM)) return; /* Type error. */
-        group = streamLookupCG(o->ptr,c->argv[2]->ptr);
-    }
-
-    /* No key or group? Send an error given that the group creation
-     * is mandatory. */
-    if (o == NULL || group == NULL) {
-        addReplyErrorFormat(c,"-NOGROUP No such key '%s' or "
-                              "consumer group '%s'", (char*)c->argv[1]->ptr,
-                              (char*)c->argv[2]->ptr);
+    if (checkType(c,o,OBJ_STREAM)) return; /* Type error. */
+    if (o == NULL) {
+        addReplyErrorFormat(c, "No such key '%s'",
```

Comment:
let's keep the old error code.

Reply:
i'd suggest to move the type check and all creations to after argument parsing.
i know it wasn't the case, but let's improve.